multichain-testing/scripts/fast-usdc-tool.ts

```diff
@@ -0,0 +1,283 @@
+    if (condition !== expected) {
+      throw new Error(
+        `Condition: ${message} failed. Expected ${expected} got ${condition}.`,
+      );
+    }
+  },
+} as ExecutionContext;
+
+// from ../test/fast-usdc/fast-usdc.test.ts
+type VStorageClient = Awaited<ReturnType<typeof commonSetup>>['vstorageClient'];
+const agoricNamesQ = (vsc: VStorageClient) =>
+  harden({
+    brands: <K extends AssetKind>(_assetKind: K) =>
+      vsc
+        .queryData('published.agoricNames.brand')
+        .then(pairs => Object.fromEntries(pairs) as Record<string, Brand<K>>),
+  });
+
+const fastLPQ = (vsc: VStorageClient) =>
+  harden({
+    metrics: () =>
+      vsc.queryData(`published.fastUsdc.poolMetrics`) as Promise<PoolMetrics>,
+    info: () =>
+      vsc.queryData(`published.${contractName}`) as Promise<{
+        poolAccount: string;
+        settlementAccount: string;
+      }>,
+  });
+
+const parseCommandLine = () => {
+  const { values, positionals } = parseArgs({
+    options: {
+      eud: {
+        type: 'string',
+      },
+      oracle: {
+        type: 'string',
+      },
+      mnemonic: {
+        type: 'string',
+      },
+      help: {
+        type: 'boolean',
+      },
+    },
+    allowPositionals: true,
+  });
+
+  if (values.help) {
+    console.log(USAGE);
+    return undefined;
+  }
+
+  const command = positionals[0];
+  const mnemonic = values.mnemonic;
+  const suppliedOracles = values.oracle?.split(',');
+  const oracles = suppliedOracles || [
+    'oracle1:agoric1yupasge4528pgkszg9v328x4faxtkldsnygwjl',
+    'oracle2:agoric1dh04lnl7epr7l4cpvqqprxvam7ewdswj7yv6ep',
+    'oracle3:agoric1ujmk0492mauq2f2vrcn7ylq3w3x55k0ap9mt2p',
+  ];
+  const eud = values.eud;
+
+  return { command, eud, mnemonic, oracles, provisionOracles: true };
+};
+
+const main = async () => {
+  const job = parseCommandLine();
+  if (!job) return undefined;
+  const { command, eud, mnemonic, oracles, provisionOracles } = job;
+  const {
+    chainInfo,
+    commonBuilderOpts,
+    deleteTestKeys,
+    faucetTools,
+    nobleTools,
+    provisionSmartWallet,
+    setupTestKeys,
+    startContract,
+    vstorageClient,
+  } = await commonSetup(runT, { config: '../config.fusdc.yaml' });
+
+  const assertProvisioned = async address => {
+    try {
+      await vstorageClient.queryData(`published.wallet.${address}.current`);
+    } catch {
+      throw new Error(`${address} is not provisioned`);
+    }
+  };
+
+  const provisionWallet = async (mnemonic: string) => {
+    // provision-one must be called by the owner, so we need to add the key to the test keyring
+    const keyname = 'temp';
+    const address = (await setupTestKeys([keyname], [mnemonic]))[keyname];
+    try {
+      await provisionSmartWallet(address, {
+        BLD: 100n,
+        IST: 100n,
+      });
+    } finally {
+      await deleteTestKeys([keyname]);
+    }
+  };
+
+  const start = async () => {
+    if (!chainInfo.noble) {
```

Comment:
It looks like the env stuff is gone (yay!), but I still can't see how `chainInfo.noble` is anything but a design-time constant.